flood_adapt/dbs_controller.py

```diff
@@ -58,6 +82,26 @@ def __init__(
             The name of the database.
         Notes
         """
+        if database_path is None or database_name is None:
+            if not self._init_done:
+                raise ValueError(
+                    """Database path and name must be provided for the first initialization. 
+                    To do this, run api_startup.read_database(database_path, site_name) first."""
+                )
+            else:
+                return  # Skip re-initialization
+
+        if (
+            self._init_done
+            and self.database_path == database_path
+            and self.database_name == database_name
+        ):
+            return  # Skip re-initialization
+
+        # If the database is not initialized, or a new path or name is provided, (re-)initialize
+        self.database_path = database_path
+        self.database_name = database_name
```

Comment:
Maybe add a warning or logging statement that the Database is being (re)initialized?